pkg/generic/thriftidl_provider.go

```diff
@@ -223,14 +326,23 @@ func (p *ThriftContentWithAbsIncludePathProvider) UpdateIDL(mainIDLPath string,
 	if !ok {
 		return fmt.Errorf("miss main IDL content for main IDL path: %s", mainIDLPath)
 	}
+	var svc *descriptor.ServiceDescriptor
 	tree, err := ParseContent(mainIDLPath, mainIDLContent, includes, true)
 	if err != nil {
 		return err
 	}
-	svc, err := thrift.Parse(tree, thrift.DefaultParseMode())
+	svc, err = thrift.Parse(tree, thrift.DefaultParseMode())
 	if err != nil {
 		return err
 	}
+
+	if p.opts.DynamicGoEnabled {
```

Comment:
also identical to L312 block?

and except for the type of p, the rest is the same as ThriftContentProvider?

Reply:
I'll refactor

Reply:
done